### PR TITLE
fix: `cy.contains()` ignores `<style>` and `<script>` without removing them.

### LINKED ISSUE #19377: `cy.contains` removes `<script>` and `<style>` nodes from the app under test
### Current behavior

Since version `9.0.0`, calling `cy.contains` removes `<script>` and `<style>` nodes from the DOM of the app under test. This is most likely caused by the change in #17477. From the description in the changelog

> `cy.contains()` no longer yields the `<body>` element when it matches the content of `<script>` or `<style>` tags.

I would expect that this only affects the result of the `contains` command, but leaves the actual DOM unchanged.

### Desired behavior

The `cy.contains` command should not alter the DOM of the application under test as a side effect.

### Test code to reproduce

The problem can be reproduced with https://github.com/simfeld/cypress-contains-script. Note that downgrading to `8.7.0` makes these tests pass.

### Cypress Version

9.0.0 and later

### Other

_No response_

### PR DESCRIPTION
**Pull request by sainthkh**
- Closes #19377

### User facing changelog

`cy.contains()` ignores `<style>` and `<script>` without removing them.

### Additional details
- Why was this change necessary? => `cy.contains()` removed `<style>` and `<script>` to ignore them. But it causes many unexpected errors. 
- What is affected by this change? => N/A
- Any implementation details to explain? => It now filters out those tags manually, not by sizzle. 

### How has the user experience changed?

N/A

### PR Tasks
- [x] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/driver/src/dom/elements/find.ts [hunk 1]
```diff
@@ -125,10 +125,16 @@ export const getFirstDeepestElement = ($el: JQuery, index = 0) => {
   const $current = $el.slice(index, index + 1)
   const $next = $el.slice(index + 1, index + 2)
 
-  if (!$next) {
+  if (!$next || $current.length === 0) {
     return $current
   }
 
+  // https://github.com/cypress-io/cypress/issues/19377
```
Comment (emilyrohrbough) [inline packages/driver/src/dom/elements/find.ts:132]: Lets link to the original issue this change was meant to solve: 
```suggestion
  // https://github.com/cypress-io/cypress/issues/14861
```

File: packages/driver/src/dom/elements/find.ts [hunk 2]
```diff
@@ -252,14 +244,12 @@ export const getContainsSelector = (text, filter = '', options: {
 
     // taken from jquery's normal contains method
     cyContainsSelector = function (elem) {
-      elem = removeScriptAndStyleElements(elem)
       let testText = normalizeWhitespaces(elem)
```
Comment (emilyrohrbough) [inline packages/driver/src/dom/elements/find.ts:247]:
```suggestion
      const testText = normalizeWhitespaces(elem)
```

File: packages/driver/cypress/integration/commands/querying_spec.js [hunk 3]
```diff
@@ -1977,6 +1978,35 @@ space
         cy.visit('fixtures/content-in-body.html')
         cy.contains('I am in the script tag in body', { timeout: 500 })
       })
+
+      // https://github.com/cypress-io/cypress/issues/19377
+      describe('cy.contains() does not remove <style> and <script> tags', () => {
+        it('cy.contains removes style tags from the DOM', () => {
```
Comment (emilyrohrbough) [inline packages/driver/cypress/integration/commands/querying_spec.js:1984]:
```suggestion
        it('cy.contains() does not remove style tags from the DOM', () => {
```

File: packages/driver/cypress/integration/commands/querying_spec.js [hunk 4]
```diff
@@ -1977,6 +1978,35 @@ space
...
+          cy.visit('fixtures/content-in-body.html')
+
+          cy.get('button#my_button_1').should('be.visible')
+          cy.contains('Hello').should('be.visible')
+          cy.get('button#my_button_1').should('be.visible')
+        })
+
+        it('cy.contains removes script tags from the DOM', () => {
```
Comment (emilyrohrbough) [inline packages/driver/cypress/integration/commands/querying_spec.js:1992]:
```suggestion
        it('cy.contains does not remove script tags from the DOM', () => {
```

**Comment by emilyrohrbough** [comment]
I pulled this change down locally & verifying the changes agains the [cypress-contains-script repo](https://github.com/simfeld/cypress-contains-script) provided to demonstrate the issue. This change appears to resolve #https://github.com/cypress-io/cypress/issues/19377 while maintain the intended fix for https://github.com/cypress-io/cypress/issues/14861.

**Review by chrisbreiding — APPROVED** [review]

**Review by chrisbreiding — APPROVED** [review]

**Review by emilyrohrbough — DISMISSED** [review]

File: packages/driver/src/dom/elements/find.ts [hunk 5]
```diff
@@ -125,10 +125,16 @@ export const getFirstDeepestElement = ($el: JQuery, index = 0) => {
   const $current = $el.slice(index, index + 1)
   const $next = $el.slice(index + 1, index + 2)
 
-  if (!$next) {
+  if (!$next || $current.length === 0) {
     return $current
   }
 
+  // https://github.com/cypress-io/cypress/issues/14861
+  // filter out <script> and <style> tags
```
Comment (mjhenkes) [inline packages/driver/src/dom/elements/find.ts:133]:
```suggestion
  // filter out the <script> and <style> tags
```

**Review by chrisbreiding — APPROVED** [review]

**Review by emilyrohrbough — APPROVED** [review]

**Comment by emilyrohrbough** [comment]
Thank you @sainthkh for the quick fix!

**Comment by simfeld** [comment]
Thanks @sainthkh and Cypress team for the quick response, much appreciated!

**Pull request merged**